solr/contrib/ltr/src/test/org/apache/solr/ltr/TestLTROnSolrCloud.java

```diff
@@ -70,6 +72,63 @@ public void tearDown() throws Exception {
     super.tearDown();
   }
 
+  @Test
+  public void testSimpleQueryCustomSort() throws Exception {
+    SolrQuery query = new SolrQuery("*:*");
+    query.setRequestHandler("/query");
+    query.setFields("*,score,[shard]");
```

Comment:
Please excuse me if it's too early to bring up also since you already mentioned the _"The current solution only works if the original sort did not sort by score. ..."_ qualification elsewhere in the pull request.

Have you considered

```suggestion
    query.setFields("*,[shard]");
```

i.e. the federator likely needing to ask the shard for the `score` if/because it will be needed but the solr client might not have requested it in their request?

And again, perhaps too early to bring up, but since re-ranking is not limited to LTR it would be great to eventually have test coverage in `solr/core` (though I don't know where might be a good fit, going by test class names only DistributedQueryComponentCustomSortTest might be a match).

What do you think?

Reply:
No worries, thank you for pointing out the fl, we did not consider that.

>  query.setFields("*,score,[shard]");
query.setFields("*,[shard]");

That change actually broke the tests. We will look for a solution to that.

I think tests in the core package are a good idea. Instead of expanding `DistributedQueryComponentCustomSortTest` an approach could be to add a new `DistributedQueryComponentReRankTest` with `ShardsRepeat` instead of `ShardsFixed`.
That way we could separate the two use cases of custom sort and reranking.

Reply:
We fixed it by adding the `GET_SCORES` flag to the rb when reRanking is enabled. (see f3ccac9)